linker/Mono.Linker.Steps/ResolveFromAssemblyStep.cs

```diff
@@ -82,8 +82,31 @@ public static void ProcessLibrary (LinkContext context, AssemblyDefinition assem
 			context.Annotations.Push (assembly);
 
 			foreach (TypeDefinition type in assembly.MainModule.Types)
-				MarkType (context, type);
-
+				MarkType (context, type);
+
+			if (assembly.MainModule.HasExportedTypes) {
+				foreach (var exported in assembly.MainModule.ExportedTypes) {
+					bool isForwarder = exported.IsForwarder;
+					var declaringType = exported.DeclaringType;
+					while (!isForwarder && (declaringType != null)) {
+						isForwarder = declaringType.IsForwarder;
+						declaringType = declaringType.DeclaringType;
+					}
+
+					if (!isForwarder)
+						continue;
+					var resolvedExportedType = exported.Resolve ();
+					var resolvedTypeAssembly = context.Resolve (resolvedExportedType.Scope);
+					context.Annotations.Push (resolvedTypeAssembly);

```

Comment:
If a library assembly is specified as a root via -a, that means that anything in the library can be accessed. I think exported types should be included for consistency. For example, I may specify -a for an assembly because I believe anything in it can potentially be a target of a reflection. That should include exported types.